app/controllers/web_controller.rb

```diff
@@ -4,15 +4,18 @@ def index
 
   def data
     sanitized_search = I18n.transliterate(params[:q]).gsub(/(\W)/, " ")
-    @artists = Search.json_for(sanitized_search)
-    @related_artists = Search.json_for_related(sanitized_search)
+    @artist = Search.for(sanitized_search)[0]
+    children_artists = @artist.related_artists
+    grandchildren_artists = children_artists.map do |artist|
+                                artist.related_artists.delete_if {|artist| artist.name == @artist.name}
```

Comment:
Shadowing outer local variable - `artist`.<br>Line is too long. [103/80]<br>Space between { and | missing.<br>Space missing inside }.